src/gluonts/mx/distribution/inflated_beta.py

```diff
@@ -0,0 +1,258 @@
+    F
+    """
+    is_reparameterizable = False
+
+    @validated()
+    def __init__(
+        self, alpha: Tensor, beta: Tensor, zero_probability: Tensor
+    ) -> None:
+        super().__init__(
+            alpha=alpha,
+            beta=beta,
+            zero_probability=zero_probability,
+            one_probability=alpha.zeros_like(),
+        )
+
+
+class OneInflatedBeta(ZeroAndOneInflatedBeta):
+    r"""
+    One Inflated Beta distribution as in Raydonal Ospina, Silvia L.P. Ferrari: Inflated Beta Distributions
+
+    Parameters
+    ----------
+    alpha
+        Tensor containing the alpha shape parameters, of shape `(*batch_shape, *event_shape)`.
+    beta
+        Tensor containing the beta shape parameters, of shape `(*batch_shape, *event_shape)`.
+    one_probability
+        Tensor containing the probability of ones, of shape `(*batch_shape, *event_shape)`.
+    F
+    """
+    is_reparameterizable = False
+
+    @validated()
+    def __init__(
+        self, alpha: Tensor, beta: Tensor, one_probability: Tensor
+    ) -> None:
+        super().__init__(
+            alpha=alpha,
+            beta=beta,
+            zero_probability=alpha.zeros_like(),
+            one_probability=one_probability,
+        )
+
+
+class ZeroAndOneInflatedBetaOutput(DistributionOutput):
+    args_dim: Dict[str, int] = {
+        "alpha": 1,
+        "beta": 1,
+        "zero_probability": 1,
+        "one_probability": 1,
+    }
+    distr_cls: type = ZeroAndOneInflatedBeta
+
+    @classmethod
+    def domain_map(cls, F, alpha, beta, zero_probability, one_probability):
+        r"""
+        Maps raw tensors to valid arguments for constructing a ZeroAndOneInflatedBeta
+        distribution.
+
+        Parameters
+        ----------
+        F:
+        alpha:
+            Tensor of shape `(*batch_shape, 1)`
+        beta:
+            Tensor of shape `(*batch_shape, 1)`
+        zero_probability:
+            Tensor of shape `(*batch_shape, 1)`
+
+        Returns
+        -------
+        Tuple[Tensor, Tensor]:
+            Four squeezed tensors, of shape `(*batch_shape)`: First two have entries mapped to the
+            positive orthant, zero_probability is mapped to (0, 1), one_probability is mapped to (0, 1-zero_probability)
+        """
+        epsilon = np.finfo(cls._dtype).eps  # machine epsilon
+
+        alpha = softplus(F, alpha) + epsilon
+        beta = softplus(F, beta) + epsilon
+        zero_probability = F.sigmoid(zero_probability)
+        one_probability = (1 - zero_probability) * F.sigmoid(one_probability)
+        return (
+            alpha.squeeze(axis=-1),
+            beta.squeeze(axis=-1),
+            zero_probability.squeeze(axis=-1),
+            one_probability.squeeze(axis=-1),
+        )
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def value_in_support(self) -> float:
+        return 0.5
+
+
+class ZeroInflatedBetaOutput(DistributionOutput):
```

Comment:
Couldn't this inherit from `ZeroAndOneInflatedBetaOutput` so as to avoid some repetitions? Like the `event_shape` and `value_in_support` methods could be omitted that way